### PR TITLE
[r314] Put an hard-coded cap on the pooled TimeSeries labels, samples and histograms slices

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

**DO NOT MERGE**. This is just an experiment.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by NickAnge — APPROVED** [review]

**Review by duricanikolic — APPROVED** [review]
LGTM

File: pkg/mimirpb/timeseries.go [hunk 1]
```diff
@@ -19,11 +19,14 @@ import (
...
-	minPreallocatedTimeseries         = 100
-	minPreallocatedLabels             = 20
-	minPreallocatedSamplesPerSeries   = 10
-	minPreallocatedExemplarsPerSeries = 1
-	maxPreallocatedExemplarsPerSeries = 10
+	minPreallocatedTimeseries          = 100
+	minPreallocatedLabels              = 20
+	maxPreallocatedLabels              = 100
+	minPreallocatedSamplesPerSeries    = 10
+	maxPreallocatedSamplesPerSeries    = 50
```
Comment (colega) [inline pkg/mimirpb/timeseries.go:26]: Not sure if this is too aggressive, it won't make a difference if this is 10x of that in terms of memory usage, but I also don't have any data to support this.


(If I was writing this code, I'd do 1024 here or something like that)

**Review by colega — APPROVED** [review]

File: pkg/mimirpb/timeseries.go (hunk 1 above)
Comment (pracucci) [reply pkg/mimirpb/timeseries.go:26]: In the Prometheus world. We typically expect 1 sample per series. When remote write is catching up we could get multiple samples. For example, assuming 4DPM, 50 samples = 11 minutes of data. It still looks a lot coming from a single WriteRequest, considering how WAL is replayed.

What's the legit case you have in mind for 1K samples for 1 series in 1 request?

**Comment by pracucci** [comment]
See https://github.com/grafana/mimir/pull/10040 instead.

**Pull request closed without merging**